pkg/patterns/declarative/pkg/applier/applylib.go

```diff
@@ -54,9 +54,13 @@ func (a *ApplySetApplier) Apply(ctx context.Context, opt ApplierOptions) error {
 
 	patchOptions.Force = &opt.Force
 
-	dynamicClient, err := dynamic.NewForConfig(opt.RESTConfig)
-	if err != nil {
-		return fmt.Errorf("error building dynamic client: %w", err)
+	dynamicClient := opt.DynamicClient
+	if dynamicClient == nil {
+		d, err := dynamic.NewForConfig(opt.RESTConfig)
+		if err != nil {
+			return fmt.Errorf("error building dynamic client: %w", err)
+		}
+		dynamicClient = d
```

Comment:
Any reason not to set opt.DynamicClient to d in this case? Could save us need to repeatedly generating dynamicClients during repeated reconciles.

Reply:
Good question, I'll take a look!  I don't _think_ dynamic client caches any server-generated state, so it's "just memory", but I'll chekc!